toolchain/semantics/semantics_context.cpp

```diff
@@ -379,6 +379,44 @@ auto SemanticsContext::ImplicitAsImpl(SemanticsNodeId value_id,
     // Type doesn't need to change.
     return ImplicitAsKind::Identical;
   }
+
+  auto array_type = semantics_ir_->GetTypeAllowBuiltinTypes(as_type_id);
+  auto array_node = semantics_ir_->GetNode(array_type);
+  if (array_node.kind() == SemanticsNodeKind::ArrayType) {
```

Comment:
```suggestion
  auto as_type = semantics_ir_->GetTypeAllowBuiltinTypes(as_type_id);
  auto as_type_node = semantics_ir_->GetNode(as_type);
  if (as_type_node.kind() == SemanticsNodeKind::ArrayType) {
```
I think these names would be clearer, because we don't yet know that `as_type_id` refers to an array type here. (We might also want to use these variables for other cases later as this function grows.)